src/main/java/citadel/Citadel.java

```diff
@@ -0,0 +1,146 @@
+
+    /** The storage system used to persist tasks between sessions. */
+    private static final Storage db = new Storage("data/citadel");
+
+    /** The user interface used to interact with the user. */
+    private static final TextUI ui = new TextUI();
+
+    /**
+     * Loads tasks from the database (storage) into the application.
+     */
+    public static TaskList loadDatabase() {
+        items = db.getTasks();
+        return items;
+    }
+
+    /**
+     * Saves tasks from the application into the database (storage).
+     */
+    public static void saveDatabase() {
+        try {
+            db.saveData(items);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Generates a response for the user's chat message.
+     */
+    public String getResponse(String input) {
+        try {
+            return handleInput(input);
+        } catch (CitadelException e) {
+            return ui.printCitadelException(e);
+        } catch (DateTimeParseException e) {
+            return ui.printDateTimeParseException();
+        } catch (Exception e) {
+            return ui.printException(e);
+        }
+    }
+
+    /**
+     * Processes the user's input command and executes the corresponding task.
+     *
+     * @param input The user's input command as a string.
+     * @throws CitadelException If there is an error processing the input.
+     */
+    public static String handleInput(String input) throws CitadelException {
+        Commands command = Parser.parseCommand(input);
+
+        switch (command) {
+        case BYE:
+            return ui.printGoodbye();
+
+        case LIST:
+            return ui.printTasks(items);
+
+        case MARK:
+            return new MarkTask(input, items).run();
+
+        case UNMARK:
+            return new UnmarkTask(input, items).run();
+
+        case DELETE:
+            return new DeleteTask(input, items).run();
+
+        case DEADLINE:
+            return new HandleDeadline(input, items).run();
+
+        case EVENT:
+            return new HandleEvent(input, items).run();
+
+        case TODO:
+            return new HandleTodo(input, items).run();
+
+        case FIND:
+            return new FindTask(input, items).run();
+
+        default:
+            throw new CitadelInvalidCommandException();
+        }
+
+    }
+
+    /**
+     * The main method that starts the Citadel application.
+     * <p>
+     * This method initializes the application, loads existing tasks, and enters a loop
+     * to process user commands until the user exits the application.
+     * </p>
+     *
+     * @param args Command-line arguments (not used in this application).
+     * @throws IOException If an I/O error occurs during task storage operations.
+     */
+    public static void main(String[] args) throws IOException {
+        ui.printStart();
+        loadDatabase();  // Load tasks from storage
+        ui.printTasks(items);
+
+        while (true) {
```

Comment:
this is very similar to your previous 'getResponse' method - do consider if code re-use here would help to reduce repeated code.